### PR TITLE
support integer enum

### PR DESCRIPTION
**Pull request by soundTricker**
Currently,  Enum type is created as string enum regardless of enum type.
This PR add supporting integer type.

### CODE REVIEW

**Comment by fazpu** [comment]
Thank you for the PR, please check the failing test. Thanks.

**Comment by soundTricker** [comment]
@fazpu this fail depends on #68 . It will get success when #68 will merge.

File: demo-app/client/generated/defs/Goods.ts [hunk 1]
```diff
@@ -14,7 +14,19 @@ export interface Goods {
...
   currency: CurrencyGoodsEnum;
+  status?: StatusGoodsEnum;
+  refStatus?: RefStatusGoodsEnum;
 }
 
 export type CurrencyGoodsEnum =
   'CZK';
+
+export type StatusGoodsEnum =
+  1|
```
Comment (jnwltr) [inline demo-app/client/generated/defs/Goods.ts]: Can you, please, add space before `|`?

File: demo-app/client/swagger-files/unit-test-swagger.json [hunk 2]
```diff
@@ -821,6 +821,19 @@
           "enum": [
             "CZK"
           ]
+        },
+        "status": {
+          "title": "Status",
+          "type": "integer",
+          "enum": [
+            1,2,3
```
Comment (jnwltr) [inline demo-app/client/swagger-files/unit-test-swagger.json]: Can you, please, add space after `,`?

File: src/common.ts [hunk 3]
```diff
@@ -33,8 +33,28 @@ export function processProperty(prop: Schema, name = '', namespace = '',
 
     const list = prop.enum || prop.items.enum;
     const exp = exportEnums ? 'export ' : '';
-    enumDeclaration = `${exp}type ${type} =\n` + indent('\'' + list.join('\' |\n\'')) + '\';';
-
+    console.log(prop.type);
+
+    const generateEnum = (originType: string, exp: string, type: string, list: any[], $ref: string): string => {
+      console.log(originType);
```
Comment (jnwltr) [inline src/common.ts]: Please, remove.

File: src/common.ts [hunk 4]
```diff
@@ -33,8 +33,28 @@ export function processProperty(prop: Schema, name = '', namespace = '',
 
     const list = prop.enum || prop.items.enum;
     const exp = exportEnums ? 'export ' : '';
-    enumDeclaration = `${exp}type ${type} =\n` + indent('\'' + list.join('\' |\n\'')) + '\';';
-
+    console.log(prop.type);
```
Comment (jnwltr) [inline src/common.ts]: Please, remove.

**Comment by jnwltr** [comment]
Thanks. Please, run `npm run lint` to check problems.

File: src/common.ts [hunk 5]
```diff
@@ -33,8 +33,28 @@ export function processProperty(prop: Schema, name = '', namespace = '',
...
+      switch(originType) {
+        case undefined:
+          const defType = translateType($ref);
+          return generateEnum(defType.type, exp, type, list, undefined);
+        case 'integer':
+          return `${exp}type ${type} =\n` + indent(list.join('|\n')) + ';';
+        case 'string':
+          return `${exp}type ${type} =\n` + indent('\'' + list.join('\' |\n\'')) + '\';';
+        default:
+        if (typeof list[0] === 'number') {
```
Comment (jnwltr) [inline src/common.ts]: Please, indent.

File: src/common.ts [hunk 6]
```diff
@@ -33,8 +33,28 @@ export function processProperty(prop: Schema, name = '', namespace = '',
...
+          return `${exp}type ${type} =\n` + indent(list.join('|\n')) + ';';
+        } else {
+          return `${exp}type ${type} =\n` + indent('\'' + list.join('\' |\n\'')) + '\';';
+        }
+      }
+    } 
+    enumDeclaration = generateEnum(prop.type, exp, type, list, prop.$ref);
```
Comment (jnwltr) [inline src/common.ts]: Anyway, I would imagine a bit simpler modification that takes care of (diverts) inner `list.join` only. Please, try to simplify.

File: src/types.ts [hunk 7]
```diff
@@ -23,7 +23,7 @@ interface ParameterSchemaBase {
   allowEmptyValue?: boolean;
   default?: any;
   description?: string;
-  enum?: string[];
+  enum?: string[]|number[];
```
Comment (jnwltr) [inline src/types.ts]: Please, add spaces around `|`.

**Comment by soundTricker** [comment]
@jnwltr sorry my bad code. I've fixed my code, please re-review it :)

**Comment by fazpu** [comment]
The build is failing...

**Comment by soundTricker** [comment]
this build failing is happened by [this issue](https://github.com/jnwltr/swagger-angular-generator/pull/68).

**Comment by jnwltr** [comment]
@soundTricker, please, rebase and we can merge. Thx

**Comment by soundTricker** [comment]
@jnwltr Rebase & Pushed :)

**Pull request merged**